### ISSUE TITLE
Initiate components in separate try/catch blocks

### ISSUE THREAD

**Issue by carlmjohnson** [alpinejs/alpine#445]
Given this code:

```
<html>
  <script src="https://cdn.jsdelivr.net/gh/alpinejs/alpine@v2.x.x/dist/alpine.min.js" defer></script>
  <body>
    <div x-data="{x: 1}">
      <span x-text="x">loads</span>
    </div>
    <div x-data="foo()">
    </div>
    <div x-data="{x: 1}">
      <span x-text="x">does not load</span>
    </div>
  </body>
</html>
```

The first span gets replace by 1 and the last span does not because the middle component fails to load.

In the real world, this is a problem because you use a feature with only partial support like IntersectionObserver or fetch, it is possible for your components to fail to load in some browsers but not others, and you don't notice this during development.

I think one way to fix with would be to just run every initiation in its own try/catch block.

**Comment by SimoTod** [comment]
👍 If you have a fix in mind, feel free to drop a PR, the project is open source and any contribution is more than welcome. 

P.s. If you're using an unsupported feature without a polyfill, one or more components will still be broken so you still need to test your page on multiple browsers. 
Even with the try catch you won't see the failure during development (and it will be harder to spot since only a minimal part of your website will be broken).

**Comment by carlmjohnson** [comment]
Yeah, the idea is that I don't mind if my scroll thingie is broken as long the most popular fetcher isn't broken too. I'll see if there's anything I can contribute or it's all over my head JS magic… :-)
